lib/flight_scheduler/schedulers/backfilling_scheduler.rb

```diff
@@ -75,6 +75,9 @@ def run_allocation_loop(candidates)
       break
     end
 
+    # Works out if there are more idling nodes then the shortage
+    excess = (partition.nodes.count { |n| n.state == 'IDLE' } > backfill.shortage)
```

Comment:
This isn't quite right.  Consider the case where 2 nodes have connected to the controller.  There are 2 jobs requiring 2 nodes and a third job requiring 3 nodes.  When the second job is completed, the shortage is calculated as `1`, the number of idle nodes is calculated as `2`.  This is determined to be an excess of nodes and so an `excess` notification is sent.  Also, shortly later, the third job is set as pending on `Resources`, so an `insufficient` notification is sent.

As the backfilling is currently changing significantly, getting this correct right now is probably not going to be correct when my changes  land.  We should change this to, "if there are zero jobs pending on resources and there are idle nodes we have an excess".  When my backfilling changes land, it can be looked at again then.

Reply:
https://github.com/openflighthpc/flight-scheduler-controller/commit/7e2b132082203df553a1e162e4f0d2e864539156